### ISSUE TITLE
[Accuracy]fp32 model result is WRONG when marking some intermediate tensor as output, while the result is OK when not marking

### ISSUE THREAD

**Issue by FdyCN** [NVIDIA/TensorRT#3258]
## Description
How come??? fp32 model result is OK, BUT when I mark some intermediate tensor as model output, fp32 model result is wrong!

When I try to dump intermediate tensor in TensorRT， according to some early issue. I tried to mark output in ONNX-TensorRT conversions. insert codes into [this potison](https://github.com/onnx/onnx-tensorrt/blob/0462dc31ae78f48744b6141ae376df1f96d3f459/ModelImporter.cpp#L628)

and my codes are:
```
    for (int i = 0; i < graph.node_size(); i ++){
        ::ONNX_NAMESPACE::NodeProto const& node = graph.node(i);

        if( node.output().size() > 0 && (node.op_type() == "MatMul" || node.op_type() == "LayerNormalization" || node.op_type() == "Conv"
                                         || node.op_type() == "Softmax")) {
            nvinfer1::ITensor* new_output_tensor_ptr = &_importer_ctx.tensors().at(node.output(0)).tensor();
            new_output_tensor_ptr->setName(node.output(0).c_str());
            _importer_ctx.network()->markOutput(*new_output_tensor_ptr);
        }
    }
```

and I try to set some debug flag:
```
    nvinfer1::IBuilderConfig *config = builder->createBuilderConfig();
    config->setFlag(nvinfer1::BuilderFlag::kGPU_FALLBACK);
    config->setFlag(nvinfer1::BuilderFlag::kSPARSE_WEIGHTS);
    config->setFlag(nvinfer1::BuilderFlag::kOBEY_PRECISION_CONSTRAINTS);
    config->setFlag(nvinfer1::BuilderFlag::kENABLE_TACTIC_HEURISTIC);
    config->setFlag(nvinfer1::BuilderFlag::kDEBUG);
```

All I want is compare the results between fp32 and fp16
## Environment

**TensorRT Version**: 8.5.1.7

**NVIDIA GPU**: 4070

**NVIDIA Driver Version**: 536.25

**CUDA Version**: 11.8

**CUDNN Version**:8.9.1

Operating System: Windows 11

Python Version (if applicable):

Tensorflow Version (if applicable):

PyTorch Version (if applicable):

Baremetal or Container (if so, version):

## Relevant Files

**Model link**:
https://huggingface.co/microsoft/beit-base-patch16-224-pt22k-ft22k

## Steps To Reproduce

**Commands or scripts**:

**Have you tried [the latest release](https://developer.nvidia.com/tensorrt)?**: yes, tested 8.5.1.7 & 8.6.1.6

**Can this model run on other frameworks?** For example run ONNX model with ONNXRuntime (`polygraphy run <model.onnx> --onnxrt`):

**Comment by FdyCN** [comment]
@ttyio  could you please take a loo at this issue？

**Comment by zerollzeng** [comment]
Mark new tensor as TRT output may break layer fusion and thus affect the final result, when you said "fp32 model result is wrong!", how is this happen exactly? you see a big diff in the final output?

**Comment by yank666** [comment]
i got some similar issue https://github.com/NVIDIA/TensorRT/issues/3313

**Comment by ttyio** [comment]
mark tensor as TRT output has 2 impacts:
- the operator that output this tensor would output fp32 by default, while it is fp16 or int8 if it is intermediate tensor.
- the fusion is broken, and we use different kernels because of the fusion broken.

@FdyCN  To compare layer by layer, need first inspect the verbose build log, and only mark those nodes at the boundary of the fused layers as network output. Thanks!

**Comment by ttyio** [comment]
closing since no activity for more than 3 weeks, pls reopen if you still have question. thanks all!
